classifier.py

```diff
@@ -0,0 +1,561 @@
+
+        num_train_optimization_steps = int(len(train_examples) / args.train_batch_size / args.gradient_accumulation_steps) * args.num_train_epochs
+        if args.local_rank != -1:
+            num_train_optimization_steps = num_train_optimization_steps // torch.distributed.get_world_size()
+
+
+        optimizer = get_optimizer(model, args.learning_rate, args.warmup_proportion, num_train_optimization_steps)
+        
+        train_dataloader = get_train_dataloader(
+            train_examples, 
+            tokenizer, 
+            label_list, 
+            args.train_batch_size, 
+            args.max_seq_length, 
+            num_train_optimization_steps,
+            args.local_rank
+        ) 
+
+        step_tracker = train_model(model, optimizer, train_dataloader, args.num_train_epochs, args.gradient_accumulation_steps, device, n_gpu)
+
+        save_model(model, args, num_labels)
+    
+    # model = load_model(args, num_labels, device) # TODO does this work as expected?
+
+    if (args.local_rank == -1 or torch.distributed.get_rank() == 0) and not args.skip_eval:
+        eval_dataloader = get_eval_dataloader(
+            processor, 
+            args.data_dir,
+            args.num_test_examples, 
+            args.eval_batch_size, 
+            args.max_seq_length, 
+            tokenizer,
+            args.seed
+        )
+        result = eval_model(model, device, eval_dataloader, step_tracker)
+        record_result(args, result)
+
+
+
+def train_model(model, optimizer, train_dataloader, num_train_epochs, gradient_accumulation_steps, device, n_gpu):
+    """ Trains the model on each example in the example set. """
+    model.train()
+    global_step = 0
+    for _ in trange(int(num_train_epochs), desc="Epoch"):
+        tr_loss = 0
+        nb_tr_examples, nb_tr_steps = 0, 0
+        for step, batch in enumerate(tqdm(train_dataloader, desc="Iteration")):
+                batch = tuple(t.to(device) for t in batch)
+                input_ids, input_mask, segment_ids, label_ids = batch
+                loss = model(input_ids, segment_ids, input_mask, label_ids)
+                if n_gpu > 1:
+                    loss = loss.mean() # mean() to average on multi-gpu.
+                if gradient_accumulation_steps > 1:
+                    loss = loss / gradient_accumulation_steps
+
+                loss.backward()
+
+                tr_loss += loss.item()
+                nb_tr_examples += input_ids.size(0)
+                nb_tr_steps += 1
+                if (step + 1) % gradient_accumulation_steps == 0:
+                    optimizer.step()
+                    optimizer.zero_grad()
+                    global_step += 1 
+    step_tracker = StepValues(
+        global_step=global_step,
+        nb_tr_steps=nb_tr_steps, 
+        tr_loss=tr_loss
+    )
+    # logger.info("   *****  STEP TRACKER *****  ")
```

Comment:
Instead of commenting logging lines out, you can decrease the level they're logged at. So you `logging.debug`.